python/pyiceberg/table/__init__.py

```diff
@@ -841,3 +868,210 @@ def to_ray(self) -> ray.data.dataset.Dataset:
         import ray
 
         return ray.data.from_arrow(self.to_arrow())
+
+
+class _SchemaUpdate(UpdateSchema):
+    def __init__(self, schema: Schema, table: Optional[Table] = None, last_column_id: Optional[int] = None):
```

Comment:
Looking at the code, the `table` is always passed in, so maybe better to make that one required
```suggestion
    def __init__(self, schema: Schema, table: Table, last_column_id: Optional[int] = None):
```

Reply:
Currently in the test, no `table` are passed in